main.go

```diff
@@ -39,10 +65,33 @@ func main() {
 }
 
 func getAllGroups(c *echo.Context) error {
-    res := ResponseTemplate{}
-    groupSlice := []Group{{1, "Foo"}, {2, "Bar"}}
+    var groups Groups
+
+    rC := RedisConnection()
+    defer rC.Close()
+
+    groupKeys, err := rC.Do("KEYS", "group:*")
+    ErrorHandler(err)
 
-    res.Content = groupSlice
+    for _, k := range groupKeys.([]interface{}) {
+        var group Group
+
+        result, err := rC.Do("GET", k.([]byte))
+        ErrorHandler(err) // TODO: handle error here or return error code?
+
+        if err := json.Unmarshal(result.([]byte), &group); err != nil {
+            ErrorHandler(err)
+        }
+        groups = append(groups, group)
+    }
+
+    res := ResponseTemplate{} // TODO: DRY out repetition of setting response values
+    res.Content = groups
+    res.ErrorCode = 0
+    res.ErrorText = "No Error"
+    res.StatusCode = http.StatusOK
+    res.StatusText = "OK"
+    res.Success = true
```

Comment:
Any alternative to setting these values like this in a request? 
Also wondering here, should this API return the error in its JSON or opt for a `panic(err)` call as I've done here now. The more I think about it, I'm imagining the former is more appropriate - just wondering in what cases to opt for the latter.

TODO: Can probably extract the Redis Access logic to own method like `func GetAllGroups` and have this one just contain logic for handling the request.